### PR TITLE
Use SiStripApvSimulationParameters from EventSetup

### PR DESCRIPTION
**Pull request by pieterdavid**
Digitizer part of https://github.com/EmyrClement/cmssw/pull/3, to be combined with https://github.com/cms-sw/cmssw/pull/27941

### CODE REVIEW

File: SimTracker/SiStripDigitizer/plugins/SiStripDigitizerAlgorithm.cc [hunk 1]
```diff
@@ -467,7 +343,12 @@ void SiStripDigitizerAlgorithm::digitize(edm::DetSet<SiStripDigi>& outdigi,
           double stripCharge = detAmpl[strip] * apv_fCPerElectron;
 
           // Get APV baseline
-          double baselineV = apvBaselineDistribution->GetRandom();
+          double baselineV = 0;
+          if (SubDet == 3) {
```
Comment (mmusich) [inline SimTracker/SiStripDigitizer/plugins/SiStripDigitizerAlgorithm.cc]: super-minor point, but can  `SiStripSubdetector::TIB` be used here?

File: SimTracker/SiStripDigitizer/plugins/SiStripDigitizerAlgorithm.cc [hunk 2]
```diff
@@ -467,7 +343,12 @@ void SiStripDigitizerAlgorithm::digitize(edm::DetSet<SiStripDigi>& outdigi,
...
+            baselineV = apvSimulationParametersHandle->sampleTIB(tTopo->tibLayer(detId), detSet_z, nTruePU_, engine);
+          } else if (SubDet == 5) {
```
Comment (mmusich) [inline SimTracker/SiStripDigitizer/plugins/SiStripDigitizerAlgorithm.cc]: and `SiStripSubdetector::TOB` here?

**Review by mmusich — COMMENTED** [review]
Just a couple of aesthetical comments

**Comment by EmyrClement** [comment]
@pieterdavid thanks for this!  I will test these two PRs together (along with the more cosmetic changes for my original PR) to check I get the same output in the validation plots I produce.

**Comment by EmyrClement** [comment]
I ran with these changes and with cms-sw/cmssw#27941, and I get hte same outcome.  You can see the cluster multiplicity vs PU [here](https://drive.google.com/drive/folders/1KrdMIYy9RPTPEaaRv_4cOK3RBlOHcfCO?usp=sharing), so it looks like the APV baseline distributions are implemented and sampled correctly with the new code.

I am happy to merge this now - any objections?

**Comment by pieterdavid** [comment]
Great to hear that, thanks for checking! I don't have anything else for now, let's see what the next step of review of the CMSSW PRs brings (the only thing that is still pending, and which probably needs something on the C++ side, is the fact that we want this for only about half of the 2016 statistics, but we can first collect some feedback on the current implementation).

**Comment by mmusich** [comment]
> the only thing that is still pending, and which probably needs something on the C++ side, is the fact that we want this for only about half of the 2016 statistics.

One needs to calculate the actual luminosity affected by the issue and use that to throw a dice in order to use the feature lumi_affected/total_lumi cases. I would configure that number either from the DB object into another member maybe just in the python configuration.

**Pull request merged**